src/components/Facility/DischargedPatientsList.tsx

```diff
@@ -233,6 +234,24 @@ const DischargedPatientsList = ({
     return days > 0 && days <= 7 && days !== 0;
   });
 
+  const queryParams = {
+    ...params,
+    csv: true,
+  };
+  const pathParams = { facility_external_id };
+
+  const { refetch: exportAsCSV } = useQuery({
+    queryKey: ["Discharged Patients", queryParams],
+    queryFn: async () => {
+      const data = await query(routes.listFacilityDischargedPatients, {
+        queryParams,
+        pathParams,
+      });
+      return data;
+    },
+    enabled: false,
+  });
```

Comment:
refer docs... query returns a callback not the data. also you are not passing the abort controller this way.

also dependencies need not be listed, since it's always manually trigerred. you can simplify things here.

```suggestion
    queryKey: ["discharged-patients-csv"],
    queryFn: query(routes.listFacilityDischargedPatients, {
        queryParams: { ...params, csv: true },
        pathParams: { facility_external_id },
      }),
    enabled: false,
  });
```